### ISSUE TITLE
load headers in right spot for all versions of clang?

### ISSUE THREAD

**Issue by beckermr** [conda-forge/openmp-feedstock#24]
Right now, if I have clang 4.0.1 in my env, but am using llvm-openmp 8.0.1, the compiler won't be able to find its headers at `$PREFIX/lib/clang/4.0.1/include/omp.h` because the version numbers differ. (The llvm-openmp package has them at `$PREFIX/lib/clang/8.0.1/include/omp.h` in this case.)

We should either force the versions to match or we need to link them properly somehow.

**Comment by beckermr** [comment]
@isuruf @jakirkham for viz

This is why the xgboost build with cmake was failing.

**Comment by beckermr** [comment]
I think adding `run_excluded` with `clang != <llvm-openmp version>` would do the trick, but IDK if we wan the constraint.

**Comment by isuruf** [comment]
A PR to `conda-forge-pinning` would fix it.

**Comment by beckermr** [comment]
Right, but in the latest gfortran build for osx, you pinned llvm-openmp to 8

https://github.com/conda-forge/gfortran_impl_osx-64-feedstock/blob/master/recipe/meta.yaml#L61

Do we need to globally remove all of these pinnings in conda forge?

**Comment by beckermr** [comment]
Given that we just rebuilt a bunch of packages with that pin, changing it now would require another rebuild of ~100 repos or so.

**Comment by isuruf** [comment]
Ah, yes. Usually `omp.h` will be found because it is also in `$PREFIX/include`, but in this case `-I$PREFIX/include` is not sent by cmake

**Comment by beckermr** [comment]
I actually think my comment above is wrong. I think the dep is in libgfortran, not the parent package, so I don't think the rebuild would be needed for those.

So we could

1. fix gfortran to not specify a version
2. add a global pin to 4

Then when we update pinnings, we have to make sure to move them together.

**Comment by isuruf** [comment]
We can't do 1. because a newer llvm-openmp is actually needed.

**Comment by beckermr** [comment]
:/ ok. 

Well then I think we need to adjust this repo to patch the right path with an activation script or something.

**Comment by beckermr** [comment]
And ofc @isuruf has better ideas :)

from him:

1. set CPATH
2. treat this as a bug in cmake

**Comment by feltech** [comment]
Not sure if this is related or a separate issue.  

With `clang` installed
```sh
$ ls ./lib/clang/
16
```
then
```sh
$ mamba install llvm-openmp
...
$ ls ./lib/clang/
16  16.0.6
$ find ./lib/clang/16.0.6/
./lib/clang/16.0.6/
./lib/clang/16.0.6/include
./lib/clang/16.0.6/include/omp.h
```

This is causing `FindOpenMP.cmake` to fail to locate `omp.h` because the `16.0.6` directory is not in the compiler's include search path, whereas the `16` directory is. 

This looks like a bug in the llvm-openmp package, however...

As mentioned, `omp.h` is also made available in `./include`, which `clang` doesn't search by default.  This is where my knowledge gets fuzzy, since I'm not clear on the logic of how or why default compiler include paths are configured.

**Comment by h-vetinari** [comment]
compiler-rt [uses](https://github.com/regro/libcfgraph/blob/master/artifacts/compiler-rt/conda-forge/linux-64/compiler-rt-16.0.6-h00ab1b0_0.json#L254-L274) the same pattern (i.e. 16.0.6, not 16), so I think this is intentional, and should be working.

I appreciate that you checked if there are existing open issues for you problem, but in this case it's a really tenuous relation. Please open a new issue and provide all the requested information (principally `conda list` and `conda info`).

**Comment by feltech** [comment]
> I appreciate that you checked if there are existing open issues for you problem, but in this case it's a really tenuous relation. Please open a new issue and provide all the requested information (principally `conda list` and `conda info`).

Done in https://github.com/conda-forge/openmp-feedstock/issues/99
